pkg/deploy/deploy.go

```diff
@@ -208,51 +210,117 @@ func deployComponentsParallel(ctx context.Context, components []graph.SortedComp
 	return errs
 }
 
-func deployComponent(ctx context.Context, component graph.SortedComponent, clientSet ClientSet, apis api.APIs, opts DeployConfigsOptions) []error {
+type deployGraph interface {
```

Comment:
Maybe this is an interface we can actually fulfill with `graph.ConfigGraph`? And not pull the underlying graph library directly into deploy, which you IMO reasonably argued against before.

We could implement things like Roots, Remove, etc. on this interface - which would allow us things like replacing the gonum lib with another one, without modifying deploy.

I think we just need Roots, RemoveNode and Copy methods?

Reply:
simple.DirectedGraph is still in the deploy, and we'd need to alias `graph.Node`, `graph.Copy`, and `simple.NewDirectedGraph` as well. 
For now, I replaced the interface with the ConfigGraph as suggested and added the NodeRemover interface to the ConfigGraph.